packages/sync-actions/src/utils/action-map-custom.js

```diff
@@ -0,0 +1,62 @@
+import * as diffpatcher from './diffpatcher'
+
+const Actions = {
+  setCustomType: 'setCustomType',
+  setCustomField: 'setCustomField',
+}
+
+const hasCustomFields = diff => Array.isArray(diff.custom)
```

Comment:
The following little function make the code below shorter while describing what happens in each case.